contracts/extensions/auto-alex-v3-endpoint.clar

```diff
@@ -0,0 +1,294 @@
+      (check-cycle (asserts! (> redeem-cycle current-cycle) err-no-redeem-revoke))
+      (tokens (mul-down (get-shares-to-tokens-per-cycle-or-default (- current-cycle u1)) (get amount request-details)))
+      (updated-request-details (merge request-details { status: (get-revoked) })))
+    (asserts! (is-eq tx-sender (get requested-by request-details)) err-unauthorised)
+    (asserts! (is-eq (get-pending) (get status request-details)) err-request-finalized-or-revoked)
+    (as-contract (try! (contract-call? .auto-alex-v3 transfer-token .auto-alex-v3 tokens (get requested-by request-details))))
+    (as-contract (try! (contract-call? .auto-alex-v3-registry set-redeem-shares-per-cycle redeem-cycle (- (get-redeem-shares-per-cycle-or-default redeem-cycle) (get amount request-details)))))
+    (print { notification: "revoke-redeem", payload: updated-request-details })
+    (as-contract (try! (contract-call? .auto-alex-v3-registry set-redeem-request request-id updated-request-details)))
+    (try! (rebase))
+		(ok true)))
+
+(define-public (execute (sender principal))
+	(let (
+      (current-cycle (unwrap-panic (contract-call? 'SP102V8P0F7JX67ARQ77WEA3D3CFB5XW39REDT0AM.alex-staking-v2 get-reward-cycle block-height))))
+		(try! (contract-call? .lisa-dao set-extensions (list { extension: .auto-alex-v3-endpoint, enabled: true } )))
+    (try! (contract-call? .auto-alex-v3-registry set-start-cycle current-cycle))
+    (try! (pause-create false))
+    (try! (pause-redeem false))
+		(ok true)))
+
+;; private functions
+;;
+
+(define-private (sum-claimed (claimed-response (response (tuple (entitled-token uint) (to-return uint)) uint)) (prior (response uint uint)))
+  (match prior
+    ok-value (match claimed-response claimed (ok (+ ok-value (get to-return claimed) (get entitled-token claimed))) err (err err))
+    err-value (err err-value)))
+
+(define-private (stake-tokens-iter (cycles-to-stake uint) (previous-response (response { current-cycle: uint, remaining: uint } uint)))
+  (match previous-response
+    ok-value
+    (let (
+      (reward-cycle (+ (get current-cycle ok-value) cycles-to-stake))
+      (redeeming (get-shares-to-tokens (get-redeem-shares-per-cycle-or-default reward-cycle)))
+      (returning (get to-return (get-staker-at-cycle reward-cycle)))
+      (staking (if (is-eq cycles-to-stake max-cycles)
+        (get remaining ok-value)
+        (if (> returning redeeming)
+          u0
+          (if (> (get remaining ok-value) (- redeeming returning))
+            (- redeeming returning)
+            (get remaining ok-value))))))
+      (and (> staking u0) (as-contract (try! (stake-tokens staking cycles-to-stake))))
+      (ok { current-cycle: (get current-cycle ok-value), remaining: (- (get remaining ok-value) staking) }))
+    err-value previous-response))
+
+(define-private (get-reward-cycle (burn-height uint))
+  (contract-call? 'SP102V8P0F7JX67ARQ77WEA3D3CFB5XW39REDT0AM.alex-staking-v2 get-reward-cycle burn-height))
```

Comment:
Call directly `alex-reserve-pool` everywhere where possible.

Reply:
We cannot because `alex-reserve-pool` hard-codes old ALEX token and `alex-staking-v2` then converts old ALEX token to new ALEX token.

Reply:
This is `alex-stacking-v2.get-reward-cycle`: 
``` 
(define-read-only (get-reward-cycle (burn-height uint))
  (contract-call? 'SP3K8BC0PPEVCV7NZ6QSRWPQ2JE9E5B6N3PA0KBR9.alex-reserve-pool get-reward-cycle 'SP3K8BC0PPEVCV7NZ6QSRWPQ2JE9E5B6N3PA0KBR9.age000-governance-token burn-height))
```

Therefore
`(contract-call? 'SP102V8P0F7JX67ARQ77WEA3D3CFB5XW39REDT0AM.alex-staking-v2 get-reward-cycle burn-height))`
can be simplified to 
` (contract-call? 'SP3K8BC0PPEVCV7NZ6QSRWPQ2JE9E5B6N3PA0KBR9.alex-reserve-pool get-reward-cycle 'SP3K8BC0PPEVCV7NZ6QSRWPQ2JE9E5B6N3PA0KBR9.age000-governance-token burn-height))`

Instead of calling two contract

Reply:
https://github.com/lisalab-io/liquid-stacking/pull/56/commits/389890350ed3123517d8aeaebd3222b1669c3661